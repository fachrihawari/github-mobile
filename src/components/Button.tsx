import React, { ReactNode } from "react";
import {
  Text,
  StyleProp,
  TextStyle,
  StyleSheet,
  ViewStyle,
  View
} from "react-native";

import Touchable from "./Touchable";
import { color } from "../config";

interface IButtonProps {
  onPress: Function;
  wrapperStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  children: String;
}

const style = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: color.primary,
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderRadius: 8
  },
  text: {
    color: "white",
    fontSize: 16
  }
});

function Button(props: IButtonProps) {
  return (
    <Touchable onPress={props.onPress}>
      <View style={[style.button, props.wrapperStyle]}>
        <Text style={[style.text, props.textStyle]}>{props.children}</Text>
      </View>
    </Touchable>
  );
}

export default Button;
