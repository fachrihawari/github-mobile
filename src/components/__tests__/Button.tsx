import "react-native";
import React from "react";
import render from "react-test-renderer";

import Button from "../Button";

it("render default Button", () => {
  const handlePress = jest.fn();
  const tree = render
    .create(<Button onPress={handlePress}>Awesome Button</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
