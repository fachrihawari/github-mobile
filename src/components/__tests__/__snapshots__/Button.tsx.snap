// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default Button 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0087E9",
          "borderRadius": 8,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Awesome Button
    </Text>
  </View>
</View>
`;
